File: 1788/1788.c
int maximizeTheBeautyOfTheGarden(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximizeTheBeautyOfTheGarden(root->left);
    int r = maximizeTheBeautyOfTheGarden(root->right);
    return 1 + (l > r ? l : r);
}

File: 1788/1788.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maximizeTheBeautyOfTheGarden = function(root) {
    if (!root) return 0;
    return 1 + Math.max(maximizeTheBeautyOfTheGarden(root.left), maximizeTheBeautyOfTheGarden(root.right));
};